src/tools/rustfmt/src/types.rs

```diff
@@ -484,21 +484,25 @@ fn rewrite_generic_args(
     span: Span,
 ) -> Option<String> {
     match gen_args {
-        ast::GenericArgs::AngleBracketed(ref data) if !data.args.is_empty() => {
-            let args = data
-                .args
-                .iter()
-                .map(|x| match x {
-                    ast::AngleBracketedArg::Arg(generic_arg) => {
-                        SegmentParam::from_generic_arg(generic_arg)
-                    }
-                    ast::AngleBracketedArg::Constraint(constraint) => {
-                        SegmentParam::Binding(constraint)
-                    }
-                })
-                .collect::<Vec<_>>();
+        ast::GenericArgs::AngleBracketed(ref data) => {
+            if data.args.is_empty() {
+                Some("".to_owned())
+            } else {
+                let args = data
+                    .args
+                    .iter()
+                    .map(|x| match x {
+                        ast::AngleBracketedArg::Arg(generic_arg) => {
+                            SegmentParam::from_generic_arg(generic_arg)
+                        }
+                        ast::AngleBracketedArg::Constraint(constraint) => {
+                            SegmentParam::Binding(constraint)
+                        }
+                    })
+                    .collect::<Vec<_>>();
 
-            overflow::rewrite_with_angle_brackets(context, "", args.iter(), shape, span)
+                overflow::rewrite_with_angle_brackets(context, "", args.iter(), shape, span)
+            }
```

Comment:
Would probably minimize the diff if we changed things as follows:

```rust
        ast::GenericArgs::AngleBracketed(ref data) if data.args.is_empty() => {
            Some(String::new())
        }
        ast::GenericArgs::AngleBracketed(ref data) => {
            let args = data
                .args
                .iter()
                .map(|x| match x {
                    ast::AngleBracketedArg::Arg(generic_arg) => {
                        SegmentParam::from_generic_arg(generic_arg)
                    }
                    ast::AngleBracketedArg::Constraint(constraint) => {
                        SegmentParam::Binding(constraint)
                    }
                })
                .collect::<Vec<_>>();

            overflow::rewrite_with_angle_brackets(context, "", args.iter(), shape, span)
        }
```

Reply:
Does it really matter that much? Is this code going to be conflicting with anything landing in rustfmt until the next subtree sync?

Reply:
Not it doesn't, though I thought I'd leave a note